src/node/mod.rs

```diff
@@ -566,7 +573,28 @@ mod builder {
             // Full + Light clients
             network.default_peers_set.in_peers = 25 + 100;
             let (keystore_remote, keystore) = (None, KeystoreConfig::InMemory);
-            let telemetry_endpoints = chain_spec.telemetry_endpoints().clone();
+
+            // HACK: Tricky way to add extra endpoints as we can't push into telemetry
+            // endpoints
+            let telemetry_endpoints = match chain_spec.telemetry_endpoints() {
+                Some(endpoints) => {
+                    let Ok(serde_json::Value::Array(extra_telemetry)) = serde_json::to_value(&telemetry) else {
```

Comment:
if it always return an array, why use `let Ok() else` syntax instead of `expect` or `context` ?

Reply:
`let extra_telemetry: serde_json::Value::Array = serde_json::to_value(&telemetry).expect("will always return an array");`
isn't this shorter? 
Seemed weird to have `let Ok() else` if we are sure the code won't go into `else` branch.

Reply:
ok nvm got it :D